routes/variableurl.go

```diff
@@ -0,0 +1,123 @@
+package routes
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	parseIndex uint
+
+	ArgName string
+
+	ParseInt struct {
+		ArgName ArgName
+	}
+	ParseFloat32 struct {
+		ArgName ArgName
+	}
+	ParseFloat64 struct {
+		ArgName ArgName
+	}
+	ParseString struct {
+		ArgName ArgName
+	}
+	ParseVoid struct {
```

Comment:
There is no such thing as `void` in Go, and you’re not parsing anything. Perhaps `Skip` would be more explicit ?